components/nft/PreviewNFTs.tsx

```diff
@@ -2,18 +2,20 @@ import { FC, useEffect, useState } from 'react'
 import { Text } from '@primer/react'
 import { Nft } from 'lib/nft/interfaces'
 import NftCard from './NftCard'
+import { useRecoilValue } from 'recoil'
+import { nftsState } from 'lib/recoil-store/atoms'
 import { ArrayWrapper, FilterArrayContext } from 'contexts/ArrayWrapper'
 import { MINIMUM_NFTS_TO_SHOW } from 'lib/constants'
 
 const PreviewNFTs: FC<{
     candyMachineAccount: string
-    itemsRemaining: number
     itemsAvailable: number
-    mintedNfts: Nft[]
-}> = ({ candyMachineAccount, itemsRemaining, itemsAvailable, mintedNfts }) => {
+    itemsRemaining: number
```

Comment:
not sure it makes sense to pass `itemsRemaining` & `itemsAvailable` and not `mintedNfts`, I feel like we can improve the code around these values, when recoil, when local state, when prop

Reply:
I think how it is is a good approach: we need 100% both `itemsRemaining` and `itemsAvailable` through props (clean, easy), then the `mintedNfts` we retrieve it with the recoil hook `useRecoilValue`.
Then, whenever we upload the cache file (we will get ALL the nfts minted and unminted from that CM) we can obtain the `unmintedNft` (ALL nft from cache - `mintedNfts` from recoil) and set it to a local state.

We could skip passing the item `itemsRemaining` and `itemsAvailable` if we had the cache always uploaded, since we don't, passing these props sounds about right.

Let me know what you think of my response.

Reply:
understood 👍🏼 thanks for the explanation